src/main/java/dao/FruitDaoImpl.java

```
@@ -0,0 +1,18 @@
+package dao;
+
+import models.FruitDao;
+import services.FruitTransaction;
+import storage.Storage;
+
+public class FruitDaoImpl implements FruitDao {
+    @Override
+    public FruitTransaction getFromCsvData(String line) {
+        String[] fields = line.split(",");
+        FruitTransaction fruitTransaction = new FruitTransaction();
+        fruitTransaction.setOperation(FruitTransaction.Operation.fromCode(fields[0]));
+        fruitTransaction.setFruit(fields[1]);
+        fruitTransaction.setQuantity(Integer.parseInt(fields[2]));

```

This method does not validate the input line for null, incorrect format, or negative quantity values. It also does not handle exceptions (e.g., NumberFormatException, ArrayIndexOutOfBoundsException) and does not wrap them in a RuntimeException with an informative message. This violates checklist items:
- 'Validate and handle invalid inputs: incorrect file path, incorrect data (e.g., negative quantity), null parameters, incorrect/unmapped operation code (should cause an error).'
- 'Don’t ignore exceptions; wrap and rethrow as RuntimeException with informative messages.'